### ISSUE TITLE
Follower units of mysql-router-k8s never go active

### ISSUE THREAD

**Issue by javacruft** [canonical/mysql-router-k8s-operator#11]
### Bug Description

When deploying the mysql-router-k8s charmed operator with multiple units, only the leader unit (as identified by Juju) manages to achieve the active status with a running mysql-router service.

### To Reproduce

juju deploy --trust -n 3 --channel edge mysql-router-k8s
juju deploy --trust -n 3 --channel edge mysql-k8s
juju deploy --trust --channel yoga/edge keystone-k8s

juju relate mysql-k8s mysql-router-k8s
juju relate keystone-k8s mysql-router-k8s

### Environment

microk9s (1.25-strict/stable channel)

### Relevant log output

```shell
unit-keystone-mysql-router-1: 11:36:49 INFO juju.cmd running containerAgent [3.0.2 8bf53dc35b25145ef39051fe4136135a3dd53d5d gc go1.19.3]
unit-keystone-mysql-router-1: 11:36:49 INFO juju.cmd.containeragent.unit start "unit"
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.upgradesteps upgrade steps for 3.0.2 have already been run.
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.probehttpserver starting http server on 127.0.0.1:65301
unit-keystone-mysql-router-1: 11:36:49 INFO juju.api cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
unit-keystone-mysql-router-1: 11:36:49 INFO juju.api connection established to "wss://10.152.183.64:17070/model/b34505d0-73a8-4f01-8e5c-d7194904701e/api"
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.apicaller [b34505] "unit-keystone-mysql-router-1" successfully connected to "10.152.183.64:17070"
unit-keystone-mysql-router-2: 11:36:49 INFO juju.cmd running containerAgent [3.0.2 8bf53dc35b25145ef39051fe4136135a3dd53d5d gc go1.19.3]
unit-keystone-mysql-router-2: 11:36:49 INFO juju.cmd.containeragent.unit start "unit"
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.upgradesteps upgrade steps for 3.0.2 have already been run.
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.probehttpserver starting http server on 127.0.0.1:65301
unit-keystone-mysql-router-1: 11:36:49 INFO juju.api cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
unit-keystone-mysql-router-1: 11:36:49 INFO juju.api connection established to "wss://10.152.183.64:17070/model/b34505d0-73a8-4f01-8e5c-d7194904701e/api"
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.apicaller [b34505] "unit-keystone-mysql-router-1" successfully connected to "10.152.183.64:17070"
unit-keystone-mysql-router-2: 11:36:49 INFO juju.api cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
unit-keystone-mysql-router-2: 11:36:49 INFO juju.api connection established to "wss://10.152.183.64:17070/model/b34505d0-73a8-4f01-8e5c-d7194904701e/api"
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.apicaller [b34505] "unit-keystone-mysql-router-2" successfully connected to "10.152.183.64:17070"
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.migrationminion migration phase is now: NONE
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.logger logger worker started
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.caasupgrader abort check blocked until version event received
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.leadership keystone-mysql-router/1 promoted to leadership of keystone-mysql-router
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.caasupgrader unblocking abort check
unit-keystone-mysql-router-1: 11:36:49 WARNING juju.worker.proxyupdater unable to set snap core settings [proxy.http= proxy.https= proxy.store=]: exec: "snap": executable file not found in $PATH, output: ""
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.migrationminion migration phase is now: NONE
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.logger logger worker started
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.leadership keystone-mysql-router leadership for keystone-mysql-router/2 denied
unit-keystone-mysql-router-2: 11:36:49 WARNING juju.worker.proxyupdater unable to set snap core settings [proxy.http= proxy.https= proxy.store=]: exec: "snap": executable file not found in $PATH, output: ""
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.caasupgrader abort check blocked until version event received
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.caasupgrader unblocking abort check
unit-keystone-mysql-router-1: 11:36:49 INFO juju.agent.tools ensure jujuc symlinks in /var/lib/juju/tools/unit-keystone-mysql-router-1
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.uniter unit "keystone-mysql-router/1" started
unit-keystone-mysql-router-2: 11:36:49 INFO juju.agent.tools ensure jujuc symlinks in /var/lib/juju/tools/unit-keystone-mysql-router-2
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.uniter resuming charm install
unit-keystone-mysql-router-1: 11:36:49 INFO juju.worker.uniter.charm downloading ch:amd64/focal/mysql-router-k8s-5 from API server
unit-keystone-mysql-router-1: 11:36:49 INFO juju.downloader downloading from ch:amd64/focal/mysql-router-k8s-5
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.uniter unit "keystone-mysql-router/2" started
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.uniter resuming charm install
unit-keystone-mysql-router-2: 11:36:49 INFO juju.worker.uniter.charm downloading ch:amd64/focal/mysql-router-k8s-5 from API server
unit-keystone-mysql-router-2: 11:36:49 INFO juju.downloader downloading from ch:amd64/focal/mysql-router-k8s-5
unit-keystone-mysql-router-1: 11:36:49 INFO juju.downloader download complete ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-2: 11:36:50 INFO juju.downloader download complete ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-1: 11:36:50 INFO juju.downloader download verified ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-2: 11:36:50 INFO juju.downloader download verified ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-0: 11:36:53 INFO juju.cmd running containerAgent [3.0.2 8bf53dc35b25145ef39051fe4136135a3dd53d5d gc go1.19.3]
unit-keystone-mysql-router-0: 11:36:53 INFO juju.cmd.containeragent.unit start "unit"
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.upgradesteps upgrade steps for 3.0.2 have already been run.
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.probehttpserver starting http server on 127.0.0.1:65301
unit-keystone-mysql-router-0: 11:36:53 INFO juju.api cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
unit-keystone-mysql-router-0: 11:36:53 INFO juju.api connection established to "wss://10.152.183.64:17070/model/b34505d0-73a8-4f01-8e5c-d7194904701e/api"
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.apicaller [b34505] "unit-keystone-mysql-router-0" successfully connected to "10.152.183.64:17070"
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.migrationminion migration phase is now: NONE
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.logger logger worker started
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.leadership keystone-mysql-router leadership for keystone-mysql-router/0 denied
unit-keystone-mysql-router-0: 11:36:53 WARNING juju.worker.proxyupdater unable to set snap core settings [proxy.http= proxy.https= proxy.store=]: exec: "snap": executable file not found in $PATH, output: ""
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.caasupgrader abort check blocked until version event received
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.caasupgrader unblocking abort check
unit-keystone-mysql-router-0: 11:36:53 INFO juju.agent.tools ensure jujuc symlinks in /var/lib/juju/tools/unit-keystone-mysql-router-0
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.uniter unit "keystone-mysql-router/0" started
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.uniter resuming charm install
unit-keystone-mysql-router-0: 11:36:53 INFO juju.worker.uniter.charm downloading ch:amd64/focal/mysql-router-k8s-5 from API server
unit-keystone-mysql-router-0: 11:36:53 INFO juju.downloader downloading from ch:amd64/focal/mysql-router-k8s-5
unit-keystone-mysql-router-0: 11:36:53 INFO juju.downloader download complete ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-0: 11:36:53 INFO juju.downloader download verified ("ch:amd64/focal/mysql-router-k8s-5")
unit-keystone-mysql-router-2: 11:36:55 INFO juju.worker.uniter hooks are retried true
unit-keystone-mysql-router-2: 11:36:56 INFO juju.worker.uniter found queued "install" hook
unit-keystone-mysql-router-1: 11:36:56 INFO juju.worker.uniter hooks are retried true
unit-keystone-mysql-router-1: 11:36:56 INFO juju.worker.uniter found queued "install" hook
unit-keystone-mysql-router-0: 11:36:56 INFO juju.worker.uniter hooks are retried true
unit-keystone-mysql-router-0: 11:36:56 INFO juju.worker.uniter found queued "install" hook
unit-keystone-mysql-router-2: 11:36:57 INFO unit.keystone-mysql-router/2.juju-log Running legacy hooks/install.
unit-keystone-mysql-router-1: 11:36:57 INFO unit.keystone-mysql-router/1.juju-log Running legacy hooks/install.
unit-keystone-mysql-router-0: 11:36:57 INFO unit.keystone-mysql-router/0.juju-log Running legacy hooks/install.
unit-keystone-mysql-router-2: 11:36:58 INFO juju.worker.uniter.operation ran "install" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:36:58 INFO juju.worker.uniter.operation ran "install" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:36:58 INFO juju.worker.uniter.operation ran "install" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:36:59 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:36:59 INFO juju.worker.uniter.operation ran "database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:00 INFO juju.worker.uniter.operation ran "database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:00 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:00 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:00 INFO juju.worker.uniter.operation ran "backend-database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:00 INFO juju.worker.uniter.operation ran "backend-database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:01 INFO juju.worker.uniter.operation ran "backend-database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:01 INFO juju.worker.uniter found queued "leader-elected" hook
unit-keystone-mysql-router-2: 11:37:01 INFO juju.worker.uniter.operation ran "leader-settings-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:01 INFO juju.worker.uniter.operation ran "database-relation-created" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:02 INFO juju.worker.uniter.operation ran "leader-elected" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:02 INFO juju.worker.uniter.operation ran "leader-settings-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:02 INFO juju.worker.uniter.operation ran "mysql-router-pebble-ready" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:03 INFO juju.worker.uniter.operation ran "mysql-router-pebble-ready" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:03 INFO juju.worker.uniter.operation ran "mysql-router-pebble-ready" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:03 INFO juju.worker.uniter.operation ran "config-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:03 INFO juju.worker.uniter found queued "start" hook
unit-keystone-mysql-router-2: 11:37:04 INFO unit.keystone-mysql-router/2.juju-log Running legacy hooks/start.
unit-keystone-mysql-router-1: 11:37:04 INFO juju.worker.uniter.operation ran "config-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:04 INFO juju.worker.uniter found queued "start" hook
unit-keystone-mysql-router-0: 11:37:04 INFO juju.worker.uniter.operation ran "config-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:04 INFO juju.worker.uniter found queued "start" hook
unit-keystone-mysql-router-1: 11:37:04 INFO unit.keystone-mysql-router/1.juju-log Running legacy hooks/start.
unit-keystone-mysql-router-0: 11:37:04 INFO unit.keystone-mysql-router/0.juju-log Running legacy hooks/start.
unit-keystone-mysql-router-2: 11:37:04 INFO juju.worker.uniter.operation ran "start" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:05 INFO juju.worker.uniter.operation ran "start" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:05 INFO juju.worker.uniter.operation ran "start" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:05 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:06 INFO juju.worker.uniter.operation ran "database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:06 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:06 INFO juju.worker.uniter.operation ran "database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:07 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:07 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:07 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:08 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:08 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:08 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:08 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:09 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:09 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:09 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:10 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:10 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:10 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:11 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:11 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:11 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:11 INFO juju.worker.uniter.operation ran "database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:12 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:12 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:12 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:13 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:13 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:13 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:14 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:14 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:14 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:15 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:15 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:15 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:16 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:16 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:16 INFO juju.worker.uniter.operation ran "backend-database-relation-joined" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:16 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:17 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:23 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:23 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:23 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:27 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:27 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:27 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:28 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:28 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:28 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:28 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:29 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:29 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:37 INFO unit.keystone-mysql-router/0.juju-log backend-database:11: database created at 2022-11-30 11:37:37.884263
unit-keystone-mysql-router-2: 11:37:37 INFO unit.keystone-mysql-router/2.juju-log backend-database:11: database created at 2022-11-30 11:37:37.897890
unit-keystone-mysql-router-1: 11:37:37 INFO unit.keystone-mysql-router/1.juju-log backend-database:11: database created at 2022-11-30 11:37:37.907434
unit-keystone-mysql-router-0: 11:37:38 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:38 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:38 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:37:39 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:37:39 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:37:39 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:38:02 INFO unit.keystone-mysql-router/2.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:02.468297
unit-keystone-mysql-router-1: 11:38:02 INFO unit.keystone-mysql-router/1.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:02.477394
unit-keystone-mysql-router-0: 11:38:02 INFO unit.keystone-mysql-router/0.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:02.553478
unit-keystone-mysql-router-2: 11:38:02 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:38:02 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:38:02 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:38:26 INFO unit.keystone-mysql-router/1.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:26.156025
unit-keystone-mysql-router-2: 11:38:26 INFO unit.keystone-mysql-router/2.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:26.157238
unit-keystone-mysql-router-0: 11:38:26 INFO unit.keystone-mysql-router/0.juju-log backend-database:11: read-only-endpoints changed on 2022-11-30 11:38:26.218326
unit-keystone-mysql-router-1: 11:38:26 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:38:26 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:38:26 INFO juju.worker.uniter.operation ran "backend-database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:41:09 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:41:27 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:41:28 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:41:29 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:41:29 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:41:29 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:41:30 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:41:30 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:41:30 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:41:30 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:41:30 INFO juju.worker.uniter.operation ran "mysql-router-peers-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-0: 11:43:22 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-2: 11:43:22 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
unit-keystone-mysql-router-1: 11:43:22 INFO juju.worker.uniter.operation ran "database-relation-changed" hook (via hook dispatching script: dispatch)
```

**Comment by taurus-forever** [comment]
JFYI, reported internally as https://warthogs.atlassian.net/browse/DPE-1114

**Comment by taurus-forever** [comment]
Dear @javacruft , the issue reported here is resolved in [PR#25](https://github.com/canonical/mysql-router-k8s-operator/pull/25).
Please check it from your side. Feel free to reopen the issue if necessary!
